src/main/scala/org/constellation/p2p/Cluster.scala

```diff
@@ -443,14 +443,12 @@ class Cluster[F[_]: Concurrent: Logger: Timer: ContextShift](ipManager: IPManage
         _ <- Logger[F].info("Trying to gracefully leave the cluster")
 
         _ <- broadcastLeaveRequest()
-        _ <- setNodeState(NodeState.Leaving)
-        _ <- broadcastNodeState()
+        _ <- compareAndSet(NodeState.all, NodeState.Leaving)
 
         // TODO: use wkoszycki changes for waiting for last n snapshots
         _ <- Timer[F].sleep(dao.processingConfig.leavingStandbyTimeout.seconds)
 
-        _ <- setNodeState(NodeState.Offline)
-        _ <- broadcastNodeState()
+        _ <- compareAndSet(NodeState.all, NodeState.Offline)
```

Comment:
shouldn't we check only for transition from Leaving to Offline, not from all to Offline?

Reply:
I guess after timeout we want to force it anyway :thinking:

Reply:
I'd say that changing state to offline should be possible only from state Leaving. and other checks should be to keep leaving untouched

Reply:
well I thought we would like to wait for snapshots to be made  or was it for snapshot made by others ?

Reply:
`Leaving` state is only for being able to respond for resolving requests and finish active consensus rounds